src/plugins/github/graphql.js

```diff
@@ -340,6 +345,129 @@ function* continuationsFromReview(
   }
 }
 
+function* usedFragmentNames(selection: Selection): Iterator<string> {
```

Comment:
What is this function for? would appreciate a docstring

Reply:
Sure; happy to add one. TL;DR: It is an error to include fragment definitions that are not uesd, so this function and `requiredFragments` perform a BFS of the query graph to find the required fragments. This one in particular gets the fragments used *non-transitively* in a given selection; `requiredFragments` does the search.

Reply:
Got it. Yeah, that context should definitely be documented - nonobvious to figure out what problem they solve absent the documentation.